components/TxForm.tsx

```diff
@@ -62,6 +62,9 @@ export const TxForm = () => {
   const [tokenAddress, setTokenAddress] = useState<string>();
   const [tokenAmount, setTokenAmount] = useState<string>("");
   const [tokenDecimals, setTokenDecimals] = useState<number>();
+  const needsApproval = ethers.utils
```

Comment:
currently the approval is hardcoded for WETH. shall we address in this PR, or elsewhere?

Reply:
Yes, I can replace line 50 with tokenAddress

Reply:
Latest commit should handle this